config/config-builder/src/validator_config.rs

```diff
@@ -228,21 +231,24 @@ impl ValidatorConfig {
                 .as_ref()
                 .and_then(|config| config.publishing_option.clone()),
         );
-
-        let waypoint = if self.build_waypoint {
-            Some(db_bootstrapper::compute_genesis_waypoint::<LibraVM>(
-                genesis.clone(),
-            )?)
-        } else {
-            None
-        };
         let genesis = Some(genesis);
-
         for node in &mut nodes {
-            node.base.waypoint = waypoint;
             node.execution.genesis = genesis.clone();
         }
 
+        if self.build_waypoint {
+            let path = TempPath::new();
+            let db_rw = DbReaderWriter::new(LibraDB::new(&path));
+            let waypoint =
+                db_bootstrapper::bootstrap_db_if_empty::<LibraVM>(&db_rw, &nodes.first().unwrap())?
+                    .ok_or_else(|| format_err!("Failed to bootstrap empty DB."))?;
+            let waypoint = Some(waypoint);
```

Comment:
Why refactor the for loop stuff? Also the node config thing below is only valid for config-builder, we'll need the transaction API again later, once we get into the tools for independently building genesis / waypoints. Since that code doesn't exist, this isn't really a blocker, I can refactor later.

Reply:
Yeah just to keep only one `bootstrap_db_if_empty`. See other comment.